### PR TITLE
fix: remove kwargs in setup and better type hint

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/_internal/runner/runner.py [hunk 1]
```diff
@@ -3,45 +3,55 @@
...
     # Default to "None", returns all available GPU devices in current environment
-    gpus = attr.ib(
-        converter=_gpu_converter,
-        type=t.List[str],
-        default=None,
-    )
+    gpus: t.List[str] = attr.field(converter=_gpu_converter, default=None)
 
-    @cpu.default
+    @cpu.default  # type: ignore
```
Comment (aarnphm) [inline bentoml/_internal/runner/runner.py:38]: pyright cannot recognize special attributes under a attrs class.

File: bentoml/_internal/runner/runner.py [hunk 2]
```diff
@@ -3,45 +3,55 @@
...
             return _query_cgroup_cpu_count()
         else:
-            return float(os.cpu_count())
+            cpu_count = os.cpu_count()
+            if cpu_count is not None:
+                return float(cpu_count)
+            raise ValueError("CPU count is NoneType")
 
-    @mem.default
+    @mem.default  # type: ignore
```
Comment (aarnphm) [inline bentoml/_internal/runner/runner.py:49]: same as above

File: bentoml/_internal/runner/utils.py [hunk 3]
```diff
@@ -308,8 +310,8 @@ def _init_var() -> t.Tuple["ctypes.CDLL", t.Dict[str, "_SimpleCData[t.Any]"]]:
         res = _drv.cuInit(0)
         if res != CUDA_SUCCESS:
             _drv.cuGetErrorString(res, ctypes.byref(err))
-            logger.error(f"cuInit failed with error code {res}: {err.value.decode()}")
-        return _drv, plc
+            logger.error(f"cuInit failed with error code {res}: {err.value.decode()}")  # type: ignore
```
Comment (aarnphm) [inline bentoml/_internal/runner/utils.py:313]: `ctypes.c_char_p` is `Optional[bytes]`, and theres not really a way to define types vis `ctypes.byref`

File: bentoml/_internal/runner/utils.py [hunk 4]
```diff
@@ -17,11 +17,13 @@
 _drv = None
 
 if TYPE_CHECKING:
-    from _ctypes import _SimpleCData
+    from ctypes import _PointerLike  # type: ignore[reportPrivateUsage]
+    from ctypes import _SimpleCData  # type: ignore[reportPrivateUsage]
```
Comment (aarnphm) [inline bentoml/_internal/runner/utils.py:21]: pyright cannot recognize `_ctypes` imports since `_ctypes` is a compiled C binary

**Comment by bojiang** [comment]
@aarnphm Just pushed an magic free utils.py

pyright actually ships with build-in `typeshed` thus `ctypes` is already typed. The magic `ignores` in this PR before is caused by misusing.

File: bentoml/_internal/frameworks/onnx.py [hunk 5]
```diff
@@ -40,11 +48,31 @@
         """
     )
 
-pd = LazyLoader("pd", globals(), "pandas")  # noqa: F811
-torch = LazyLoader("torch", globals(), "torch")  # noqa: F811
-tf = LazyLoader("tf", globals(), "tensorflow")  # noqa: F811
+if sys.version_info >= (3, 8):
```
Comment (parano) [inline bentoml/_internal/frameworks/onnx.py:51]: is this necessary? 

can we do

```
_GPUProviderType = t.List[
     t.Tuple[
         "t.Literal['CUDAExecutionProvider']",
         t.Union[t.Dict[str, t.Union[int, str, bool]], str],
     ]
 ]```

File: bentoml/_internal/frameworks/onnx.py [hunk 6]
```diff
@@ -40,11 +33,37 @@
...
+if TYPE_CHECKING:
+    import numpy as np
+    import torch
+    from pandas.core.frame import DataFrame
+    from tensorflow.python.framework.ops import Tensor as TFTensor
+
+    from ..models import ModelStore
+
+    _ProviderType = t.List[t.Union[str, t.Tuple[str, t.Dict[str, t.Any]]]]
+    _GPUProviderType = t.List[
```
Comment (parano) [inline bentoml/_internal/frameworks/onnx.py:45]: 👍

**Pull request merged**